test/apps/iodemo/io_demo.cc

```diff
@@ -989,19 +1003,16 @@ class DemoClient : public P2pDemoCommon {
             VERBOSE_LOG << " <<<< iteration " << total_iter << " >>>>";
 
             wait_for_responses(opts().window_size - 1);
-            if (_status != OK) {
-                break;
-            }
-
-            connect_all((total_iter % 10) == 0);
+            connect_all(is_control_iter(total_iter));
             if (_status != OK) {
                 break;
             }
 
             if (_active_servers.empty()) {
+                double retry_iterval = opts().client_timeout / 10;
```

Comment:
maybe add new param: reconnect interval (could be in another pr)
BTW connect_all still uses client_timeout at line 956